backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeDataLoaderService.java

```diff
@@ -0,0 +1,101 @@
+package gov.cdc.usds.simplereport.api.devicetype;
+
+import static java.util.Collections.emptyList;
+
+import gov.cdc.usds.simplereport.db.model.DeviceSupportedDisease;
+import gov.cdc.usds.simplereport.db.model.DeviceTypeSpecimenTypeMapping;
+import gov.cdc.usds.simplereport.db.model.SpecimenType;
+import gov.cdc.usds.simplereport.db.model.SupportedDisease;
+import gov.cdc.usds.simplereport.db.repository.DeviceSpecimenTypeNewRepository;
+import gov.cdc.usds.simplereport.db.repository.DeviceSupportedDiseaseRepository;
+import gov.cdc.usds.simplereport.db.repository.SpecimenTypeRepository;
+import gov.cdc.usds.simplereport.service.DiseaseService;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class DeviceTypeDataLoaderService {
+
+  final DeviceSupportedDiseaseRepository deviceSupportedDiseaseRepository;
+  final DiseaseService diseaseService;
+  final DeviceSpecimenTypeNewRepository deviceSpecimenTypeNewRepository;
+  final SpecimenTypeRepository specimenTypeRepository;
+
+  Map<UUID, List<SupportedDisease>> getSupportedDiseases(Set<UUID> deviceTypeIds) {
+    // load cached supportedDisease
+    Map<UUID, SupportedDisease> supportedDiseasesMap =
+        diseaseService.getCachedSupportedDiseasesMap();
+
+    // load deviceType -> [supportedDisease]
+    List<DeviceSupportedDisease> allByDeviceTypeIdIn =
+        deviceSupportedDiseaseRepository.findAllByDeviceTypeIdIn(deviceTypeIds);
+    Map<UUID, List<DeviceSupportedDisease>> deviceTypeIdSupportedDiseaseIdsMapping =
+        allByDeviceTypeIdIn.stream()
+            .collect(Collectors.groupingBy(DeviceSupportedDisease::getDeviceTypeId));
+
+    Map<UUID, List<SupportedDisease>> found = new HashMap<>();
+
+    // this makes sure we return an empty array if no supported diseases exist
+    deviceTypeIds.forEach(deviceTypeId -> found.put(deviceTypeId, emptyList()));
```

Comment:
@georgehager apparently I misled you the other day - `List.of()` isn't the canonical way to create an empty list anymore!